libs/libarchfpga/src/physical_types.h

```diff
@@ -840,6 +853,10 @@ struct t_logical_block_type {
     std::vector<t_physical_tile_type_ptr> equivalent_tiles; ///>List of physical tiles at which one could
                                                             ///>place this type of netlist block.
 
+    std::unordered_map<int, const t_pb_graph_pin*> pb_pin_num_map; /* pb_pin_num_map[pin logical number] -> pb_graph_pin ptr} */
+    std::unordered_map<const t_pb_graph_pin*, int> pb_pin_class_map; /* pb_pin_class_map[pb_graph_pin ptr] -> class logical number */
+    std::vector<t_class> logical_class_inf; /* logical_class_inf[class_logical_number] -> class */
```

Comment:
logical classes[class_logical_number] may be better naming?

Reply:
I wanted it to be consistent with the member (class_info) defined t_physical_tile_type. If you think I don't need to consider that, I think the name you suggested is good.